intake_bluesky/core.py

```diff
@@ -889,3 +891,101 @@ def parse_handler_registry(handler_registry):
 
 intake.registry['remote-bluesky-run'] = RemoteBlueskyRun
 intake.container.container_map['bluesky-run'] = RemoteBlueskyRun
+
+
+def concat_dataarray_pages(dataarray_pages):
+    """
+    Combines a iterable of dataarray_pages to a single dataarray_page.
+
+    Parameters
+    ----------
+    dataarray_pages: Iterabile
+        An iterable of event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    Returns
+    ------
+    event_page : dict
+        A single event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    """
+    pages = list(dataarray_pages)
+    if len(pages) == 1:
+        return pages[0]
+
+    array_keys = ['seq_num', 'time', 'uid']
+    data_keys = dataarray_pages[0]['data'].keys()
+
+    return {'descriptor': pages[0]['descriptor'],
+            **{key: list(itertools.chain.from_iterable(
+                    [page[key] for page in pages])) for key in array_keys},
+            'data': {key: xarray.concat([page['data'][key] for page in pages])
+                     for key in data_keys},
+            'timestamps': {key: xarray.concat([page['timestamps'][key] for page in pages])
+                           for key in data_keys},
+            'filled': {key: xarray.concat([page['filled'][key] for page in pages])
+                       for key in data_keys}}
+
+
+def event_page_to_dataarray_page(event_page, dims=None, coords=None):
+    """
+    Converts the event_page's data, timestamps, and filled to xarray.DataArray.
+
+    Parameters
+    ----------
+    event_page: dict
+    Returns
+    ------
+    event_page : dict
+        An event_pages with xarray.dataArrays in the data,
+        timestamp, and filled fields.
+    """
+    if coords is None:
+        coords = {'time': event_page['time']}
+    if dims is None:
+        dims = ('time',)
+
+    for key in event_page['data']:
+        event_page['data'][key] = xarray.DataArray(
+                    event_page['data'][key], dims=dims, coords=coords, name=key)
+        event_page['timestamps'][key] = xarray.DataArray(
+                    event_page['timestamps'][key], dims=dims, coords=coords, name=key)
+        event_page['filled'][key] = xarray.DataArray(
+                    event_page['filled'][key], dims=dims, coords=coords, name=key)
+
+    return event_page
+
+
+def dataarray_page_to_dataset_page(dataarray_page):
+
+    """
+    Converts the dataarray_page's data, timestamps, and filled to xarray.DataSet.
+
+    Parameters
+    ----------
+    dataarray_page: dict
+    Returns
+    ------
+    dataset_page : dict
+    """
+    dataarray_page['data'] = xarray.merge(dataarray_page['data'].values())
+    dataarray_page['timestamps'] = xarray.merge(dataarray_page['timestamps'].values())
+    dataarray_page['filled'] = xarray.merge(dataarray_page['filled'].values())
+
+    return dataarray_page
+
+
+class DaskFiller(event_model.Filler):
+    def event_page(self, doc):
+
+        @dask.delayed
```

Comment:
Remember that we need to use `dask.array.from_delayed` instead of the `@dask.delayed` decorator so that the result successfully pretends to be an array instead of a generic, opaque "delayed" object.

Reply:
I think array.from_delayed takes a dask.delayed object: https://docs.dask.org/en/latest/array-api.html#dask.array.from_delayed. So I think you use dask.delay and dask.from_delayed together.

Reply:
I see. But what is `@dask.delayed` adding here? Isn't it sufficient to return `doc`, a dict with (where applicable) for `dask.array.from_delayed` objects inside it?

Reply:
Nevermind---I was misreading this. :sheep: